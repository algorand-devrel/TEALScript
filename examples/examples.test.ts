/* eslint-disable no-unused-expressions */
/* eslint-disable func-names */
/* eslint-disable prefer-arrow-callback */
import { artifactsTest } from '../tests/common';

artifactsTest('AMM', 'examples/amm/amm.algo.ts', 'examples/amm/tealscript_artifacts/', 'ConstantProductAMM');

['Master', 'Vault'].forEach((className) => {
  artifactsTest('ARC12', 'examples/arc12/arc12.algo.ts', 'examples/arc12/', className);
});

artifactsTest('ARC75', 'examples/arc75/arc75.algo.ts', 'examples/arc75/artifacts/', 'ARC75');

artifactsTest('Auction', 'examples/auction/auction.algo.ts', 'examples/auction/tealscript_artifacts/', 'Auction');

['FactoryCaller', 'NFTFactory'].forEach((className) => {
  artifactsTest('Inner Transactions', 'examples/itxns/itxns.algo.ts', 'examples/itxns/artifacts/', className);
});

artifactsTest('Simple', 'examples/simple/simple.algo.ts', 'examples/simple/artifacts/', 'Simple');

artifactsTest('Tuple In Box', 'examples/tuple_in_box/app.algo.ts', 'examples/tuple_in_box/tealscript_artifacts/', 'ContactsApp');

artifactsTest('Calculator', 'examples/calculator/calculator.algo.ts', 'examples/calculator/artifacts/', 'Calculator');

artifactsTest('MerkleTree', 'examples/merkle/merkle.algo.ts', 'examples/merkle/artifacts/', 'MerkleTree');
