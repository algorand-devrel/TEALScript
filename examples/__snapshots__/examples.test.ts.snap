// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Examples ConstantProductAMM Artifacts Maintains program size 1`] = `1086`;

exports[`Examples ARC75 Artifacts Maintains program size 1`] = `736`;

exports[`Examples Auction Artifacts Maintains program size 1`] = `553`;

exports[`Examples FactoryCaller Artifacts Maintains program size 1`] = `422`;

exports[`Examples NFTFactory Artifacts Maintains program size 1`] = `173`;

exports[`Examples Simple Artifacts Maintains program size 1`] = `288`;

exports[`Examples ContactsApp Artifacts Maintains program size 1`] = `635`;

exports[`Examples Calculator Artifacts Maintains program size 1`] = `196`;

exports[`Examples MerkleTree Artifacts Maintains program size 1`] = `584`;
